crypto/privateKey.go

```diff
@@ -0,0 +1,98 @@
+package crypto
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aptos-labs/aptos-go-sdk/internal/util"
+)
+
+// PrivateKeyVariant represents the type of private key
+type PrivateKeyVariant string
+
+const (
+	Ed25519   PrivateKeyVariant = "ed25519"
+	Secp256k1 PrivateKeyVariant = "secp256k1"
+)
+
+// AIP80Prefixes contains the AIP-80 compliant prefixes for each private key type
+var AIP80Prefixes = map[PrivateKeyVariant]string{
+	Ed25519:   "ed25519-priv-",
+	Secp256k1: "secp256k1-priv-",
+}
+
+// FormatPrivateKey formats a hex input to an AIP-80 compliant string
+func FormatPrivateKey(privateKey interface{}, keyType PrivateKeyVariant) (string, error) {
+	aip80Prefix := AIP80Prefixes[keyType]
+
+	var hexStr string
+	switch v := privateKey.(type) {
+	case string:
+		// If it's already a string, just ensure it's properly formatted
+		var strBytes, err = util.ParseHex(v)
+		if err != nil {
+			return "", err
+		}
+
+		// Reformat to have 0x prefix
+		hexStr = util.BytesToHex(strBytes)
+	case []byte:
+		hexStr = util.BytesToHex(v)
+	default:
+		return "", fmt.Errorf("unsupported private key type: must be string or []byte")
+	}
+
+	return fmt.Sprintf("%s%s", aip80Prefix, hexStr), nil
+}
+
+// ParseHexInput parses a hex input that may be bytes, hex string, or an AIP-80 compliant string to bytes.
+//
+// You may optionally pass in a boolean to strictly enforce AIP-80 compliance.
+func ParsePrivateKey(value interface{}, keyType PrivateKeyVariant, strict ...bool) ([]byte, error) {
```

Comment:
Can simplify  this a bit with a couple changes more go like
```suggestion
func ParsePrivateKey(value interface{}, keyType PrivateKeyVariant, strict ...bool) (out []byte, err error) {
```